code/modules/vehicles/armored/armored_weapons.dm

```diff
@@ -104,14 +107,17 @@
 		return FALSE
 	if(QDELETED(current_target))
 		return FALSE
-	if(chassis.primary_weapon == src)
-		var/dir_target_diff = get_between_angles(Get_Angle(chassis, current_target), dir2angle(chassis.turret_overlay.dir))
+	if(!(armored_weapon_flags & MODULE_FIXED_FIRE_ARC))
+		return TRUE
+	if(armored_weapon_flags & MODULE_FIXED_FIRE_ARC)
+		var/turf/source_turf = chassis.primary_weapon == src ? chassis.hitbox.get_projectile_loc(src) : get_turf(src)
```

Comment:
I'm not super knowledgeable about vehicle code but will source turf work for it? Wondering if secondary weapons are inside the mech instead of constantly being moved around.

Reply:
yes it definitely works